File: 3952-trionic-array-i/trionic-array-i.cpp
class Solution {
public:
    bool isTrionic(vector<int>& nums) {
        int n = nums.size();
        int i = 1;

        // Phase 1: strictly increasing
        while (i < n && nums[i] > nums[i - 1]) {
            i++;
        }

        // must have at least one increasing step
        if (i == 1 || i == n) return false;

        // Phase 2: strictly decreasing
        int decStart = i;
        while (i < n && nums[i] < nums[i - 1]) {
            i++;
        }

        // must have at least one decreasing step
        if (i == decStart || i == n) return false;

        // Phase 3: strictly increasing
        int inc2Start = i;
        while (i < n && nums[i] > nums[i - 1]) {
            i++;
        }

        // must reach end and have at least one step
        return (i == n && inc2Start < n);
    }
};

File: 3952-trionic-array-i/trionic-array-i.java
class Solution {
    
    public boolean isTrionic(int[] nums) {
        int n = nums.length;
        
        if (n < 4) return false; // need at least 4 elements
        
        // Try all possible p and q
        for (int p = 1; p <= n - 3; p++) {
            for (int q = p + 1; q <= n - 2; q++) {
                
                if (isIncreasing(nums, 0, p) &&
                    isDecreasing(nums, p, q) &&
                    isIncreasing(nums, q, n - 1)) {
                    return true;
                }
            }
        }
        
        return false;
    }
    
    // Check strictly increasing from start to end
    private boolean isIncreasing(int[] nums, int start, int end) {
        for (int i = start; i < end; i++) {
            if (nums[i] >= nums[i + 1]) {
                return false;
            }
        }
        return true;
    }
    
    // Check strictly decreasing from start to end
    private boolean isDecreasing(int[] nums, int start, int end) {
        for (int i = start; i < end; i++) {
            if (nums[i] <= nums[i + 1]) {
                return false;
            }
        }
        return true;
    }
}